### PR TITLE
Linear scheduler

### PR DESCRIPTION
**Pull request by rrevoid**

### CODE REVIEW

File: src/diffusion/model.py [hunk 1]
```diff
@@ -134,7 +136,10 @@ def __init__(
     def configure_optimizers(self):
         # Fully control LR from scheduler
         optimizer = torch.optim.AdamW(self.parameters(), lr=1.0, weight_decay=self.weight_decay)
-        scheduler_lambda = partial(rsqrt_with_warmup, max_lr=self.lr, min_lr=self.min_lr, warmup=self.warmup)
+        if self.decay_range is not None:
```
Comment (SpirinEgor) [inline src/diffusion/model.py]: Well, that is not good criteria for choosing a scheduler. I suggest small refactoring:
1. Let's replace all optimizer-related arguments in `__init__` with `optimizer_kwargs`. And save this dict as it is (`self.opt_kwargs = opt_kwargs`)
2. In `configure_optimizer` extract necessary values from dict (e.g. `max_lr=self.opt_kwargs["max_lr"])
3. Use`scheduler` key to extract the name of the scheduler
4. In config files move all optimizer-related arguments into a separate block
5. In main script pass parsed block directly to DiDi, i.e. `DiDi(..., opt_kwargs=config.optimizer, **config.didi)`

File: src/diffusion/model.py [hunk 2]
```diff
@@ -274,3 +279,13 @@ def rsqrt_with_warmup(step: int, max_lr: float, min_lr: float, warmup: int) -> f
     if min_lr is not None:
         lr = max(lr, min_lr)
     return lr
+
+
+def linear(step: int, max_lr: float, min_lr: float, decay_range: tuple) -> float:
+    s1, s2 = decay_range
```
Comment (SpirinEgor) [inline src/diffusion/model.py]: The decay range should be int, the number of total steps of decaying. Use warmup to specify the number of steps before starting decay.

**Review by SpirinEgor — APPROVED** [review]
